NuRadioReco/modules/efieldToVoltageConverter.py

```diff
@@ -268,32 +268,35 @@ def run(self, evt, station, det, channel_ids=None):
+                # signal arrival direction (due to refraction into the ice) and account for
+                # missing power due to the Fresnel factors.
+                if not efield_is_at_antenna:
+                    zenith_antenna, t_theta, t_phi = geo_utl.fresnel_factors_and_signal_zenith(
+                        det, sim_station, channel_id, zenith)
+                else:
+                    zenith_antenna = zenith
+                    t_theta = 1
+                    t_phi = 1
 
-                if self.__caching:
-                    if not efield_is_at_antenna:
-                        zenith_antenna, t_theta, t_phi = geo_utl.fresnel_factors_and_signal_zenith(det, sim_station, channel_id, zenith)
-                    else:
-                        zenith_antenna = zenith
-                        t_theta = 1
-                        t_phi = 1
-
-                    antenna_model = det.get_antenna_model(sim_station.get_id(), channel_id, zenith_antenna)
-                    antenna_pattern = self.__antenna_provider.load_antenna_pattern(antenna_model)
-                    ant_orient = det.get_antenna_orientation(sim_station.get_id(), channel_id)
+                # Get the antenna pattern and orientation for the current channel
+                antenna_pattern, antenna_orientation = self.get_antenna_pattern_and_orientation(
+                    det, sim_station, channel_id, zenith_antenna)
 
-                    vel_tmp = self._get_cached_antenna_response(antenna_pattern, zenith_antenna, azimuth, *ant_orient)
-                    VEL = [np.array([vel_tmp['theta'] * t_theta, vel_tmp['phi'] * t_phi])]
+                # Get antenna sensitivity for the given direction
+                if self.__caching:
+                    vel = self._get_cached_antenna_response(
+                        antenna_pattern, zenith_antenna, azimuth, *antenna_orientation)
                 else:
-                    # get antenna pattern for current channel
-                    VEL = signal_processing.get_efield_antenna_factor(
-                        sim_station, ff, [channel_id], det, zenith, azimuth, self.__antenna_provider, efield_is_at_antenna)
+                    vel = antenna_pattern.get_antenna_response_vectorized(
+                        ff, zenith_antenna, azimuth, *antenna_orientation)
 
-                if VEL is None:  # this can happen if there is not signal path to the antenna
+                if vel is None:  # this can happen if there is not signal path to the antenna
                     voltage_fft = np.zeros_like(efield_fft[1])  # set voltage trace to zeros
                 else:
                     # Apply antenna response to electric field
-                    VEL = VEL[0]  # we only requested the VEL for one channel, so selecting it
-                    voltage_fft = np.sum(VEL * np.array([efield_fft[1], efield_fft[2]]), axis=0)
+                    vel = np.array([vel['theta'] * t_theta, vel['phi'] * t_phi])
```

Comment:
is it a problem if you override `vel`?

Reply:
No I do not think that this is a problem